File: C++_Programming/VirtualDemo6.cpp
#include<iostream>
using namespace std;

#pragma pack(1)
class Base 
{
    public:
        int i, j;

        void fun()                          // 1000
        {    cout<<"Base Fun\n"; }
        void gun()                          // 2000
        {    cout<<"Base Gun\n"; }
        virtual void sun()                  // 3000
        {    cout<<"Base Sun\n"; }
        virtual void run()                  // 4000
        {    cout<<"Base Run\n"; }
};  // 16 Bytes - i(4) + j(4) + 

#pragma pack(1)
class Derived : public Base 
{
    public:
        int x;

        void fun()                          // 5000
        {    cout<<"Derived Fun\n";  }
        void sun()                          // 6000
        {    cout<<"Derived Sun\n"; }
        virtual void mun()                  // 7000     
        {    cout<<"Derived Mun\n"; }
        void bun()                          // 8000
        {    cout<<"Derived Bun\n";  }


};  // 20 Bytes 


int main ()
{
    Base *BP = new Derived();
    

    BP->fun();
    BP->gun();
    BP->sun();
    BP->run();
    // BP->mun();           // Error
    // BP->bun();           // Error

    return 0;
}

File: Java_Programming/VirtualDemo6.java
class Base 
{   
        int i, j;
        void fun()                                              // 1000
        {    System.out.println("Base Fun\n"); }
        void gun()                                              // 2000
        {    System.out.println("Base Gun\n"); }
        void sun()                                              // 3000
        {    System.out.println("Base Sun\n"); }
        void run()                                              // 4000
        {    System.out.println("Base Run\n"); }
}

class Derived extends Base 
{
        int x;
        void fun()                                               // 5000
        {    System.out.println("Derived Fun\n");  }
        void sun()                                               // 6000
        {    System.out.println("Derived Sun\n"); }
        void mun()                                               // 7000     
        {    System.out.println("Derived Mun\n"); }
        void bun()                                               // 8000
        {    System.out.println("Derived Bun\n");  }
}  

class VirtualDemo6 {
    public static void main (String A[])
    {
        Base BP = new Derived();

        BP.fun();
        BP.gun();
        BP.sun();
        BP.run();
        // BP.mun();           // Error
        // BP.bun();           // Error
    }
}